go-controller/pkg/factory/factory.go

```diff
@@ -309,6 +337,21 @@ type WatchFactory struct {
 
 	iFactory  informerfactory.SharedInformerFactory
 	informers map[reflect.Type]*informer
+	ObjectCacheInterface
```

Comment:
@abhat we don't need to put ObjectCacheInterface into the WatchFactory struct; it's enough to make sure that WatchFactory implements all the methods that ObjectCacheInterface defines. A common pattern in Go is to enforce at compile time that the object (in this case WatchFactory) implements an interface that it claims to implement. That's done by something like this near the 'type WatchFactory struct {` line:

```
// WatchFactory implements the ObjectCacheInterface interface
var _ ObjectCacheInterface = &WatchFactory{}
```

Reply:
Yep, that is an inadvertent left-over from the previous implementation. I will update.